### PR TITLE
CMR-8717-2: Add search-after to OUS API with new uberjar dir structure

### PR DESCRIPTION
**Pull request by TylerHeald1**
Just the changes from the CMR-8717 branch (with some comments addressed) applied to the new directory structure used for running service-bridge with uberjar

### CODE REVIEW

**Review by eudoroolivares2016 — APPROVED** [review]

**Review by isja17 — APPROVED** [review]

**Review by dpesall — APPROVED** [review]

File: other/cmr-exchange/ous-plugin/src/cmr/ous/impl/v2_1.clj [hunk 1]
```diff
@@ -30,21 +30,24 @@
...
-        coll (collection/extract-metadata coll-promise)
-        tag-data (get-in coll [:tags (keyword collection/opendap-regex-tag) :data])
+        gran-headers (granule/extract-header-data grans-promise)
+        coll-body (collection/extract-body-metadata coll-promise)
+        tag-data (get-in coll-body [:tags (keyword collection/opendap-regex-tag) :data])
         granule-links (map granule/extract-granule-links granules)
-        service-ids (collection/extract-service-ids coll)
-        vars (common/apply-bounding-conditions endpoint token coll params)
+        sa-header (get gran-headers :cmr-search-after)
+        hits-header (get gran-headers :cmr-hits)
```
Comment (ygliuvt) [inline other/cmr-exchange/ous-plugin/src/cmr/ous/impl/v2_1.clj:38]: Thanks for fixing `cmr-hits` header as well here.

**Review by ygliuvt — CHANGES_REQUESTED** [review]
Need to update OUS API doc to document the `CMR-Search-After` feature.

The `CMR-Search-After` tests can be added together with `cmr-hits` header as part of CMR-8716.

File: other/cmr-exchange/service-bridge/resources/docs/rest-api/index.md [hunk 2]
```diff
@@ -179,12 +179,19 @@ and, the time taken to process the request.
...
+  "sa-header": "[...]",
   "items": [...],
   "request-id": ...,
   "warnings:" null
 }
 ```
 
+## Search-After
+
+CMR Service-Bridge allows the use of CMR-Search-After. When searching a collection
```
Comment (ygliuvt) [inline other/cmr-exchange/service-bridge/resources/docs/rest-api/index.md:191]: Maybe change to something like:

"CMR Service-Bridge allows the use of CMR-Search-After header to take advantage of the CMR Search-After support for paging. Any search against CMR Service Bridge that has results not fully returned in the current request will return a search-after value in the CMR-Search-After header of the search response. User can then pass this returned value in the CMR-Search-After header of the following request to retrieve the next page of result based on the specified page_size. See CMR API document on Search-After for a detailed description."

In your description, it says `the response will include the Search-After header as a header
and in the response body.`. Will the search-after header be returned in the response body also? Why do we want to do that?

Comment (TylerHeald1) [reply other/cmr-exchange/service-bridge/resources/docs/rest-api/index.md:191]: Unfortunately, with how some of the service-bridge/cmr-exchange stuff is set up, the data where the response headers are set passed through an area of code that adds values to the response body. I tried passing the data through without adding it to the response, but I don't think the code is currently set up in a good way for handling that. the headers CMR-Hits, CMR-Took, and CMR-Request-ID all go through the same thing (ending up in header and response body)

Comment (TylerHeald1) [reply other/cmr-exchange/service-bridge/resources/docs/rest-api/index.md]: With Liu's help, it is now removed from the response body

**Review by ygliuvt — APPROVED** [review]

**Pull request merged**